hera_qm/xrfi.py

```diff
@@ -607,11 +685,215 @@ def xrfi_h1c_pipe(uv, Kt=8, Kf=8, sig_init=6., sig_adj=2., px_threshold=0.2,
     else:
         return uvf_f, uvf_wf
 
+
+def xrfi_pipe(uv, alg='detrend_medfilt', Kt=8, Kf=8, xants=[], cal_mode='gain',
+              sig_init=6.0, sig_adj=2.0):
+    """xrfi excision pipeline used for H1C IDR2.2. Uses detrending and watershed algorithms above.
+    Args:
+        uv (UVData or UVCal): Object to calculate metric.
+        alg (str, optional): Algorithm for calculating metric. Default is
+            'detrend_medfilt'.
+        Kt (int, optional): Size of kernel in time dimension for detrend in
+            xrfi algorithm. Default is 8.
+        Kf (int, optional): Size of kernel in frequency dimension for detrend
+            in xrfi algorithm. Default is 8.
+        xants (list): List of antennas to flag. Default is empty list.
+        cal_mode: (str) Mode to calculate metric if uv is UVCal. Options are
+            'gain', 'chisq', and 'tot_chisq' to use the gain_array,
+            'quality_array', and 'total_quality_array', respectively.
+        sig_init (float, optional): Starting number of sigmas to flag on. Default is 6.
+        sig_adj (float, optional): Number of sigmas to flag on for data adjacent
+            to a flag. Default is 2.0.
+    Returns:
+        uvf_m (UVFlag): UVFlag object with metric after collapsing to waterfall and to
+            single pol. Weights array is set to ones.
+        uvf_fws (UVFlag): UVFlag object with flags after watershed.
+    """
+    flag_xants(uv, xants)
+    uvf_m = calculate_metric(uv, alg, Kt=Kt, Kf=Kf, cal_mode=cal_mode)
+    uvf_m.to_waterfall(keep_pol=False)
+    uvf_m.weights_array = uvf_m.weights_array.astype(np.bool).astype(np.float)
+    alg_func = algorithm_dict[alg]
+    uvf_m.metric_array[:, :, 0] = alg_func(uvf_m.metric_array[:, :, 0],
+                                           flags=~uvf_m.weights_array[:, :, 0].astype(np.bool),
+                                           Kt=Kt, Kf=Kf)
+    uvf_f = flag(uvf_m, nsig_p=sig_init)
+    uvf_fws = watershed_flag(uvf_m, uvf_f, nsig_p=sig_adj, inplace=False)
```

Comment:
Flag and watershed on each data product individually. That is, on each complete file (e.g. calibration gains), not on individual antennas/baselines.
We don't broadcast until the very end.

Reply:
I'd add this as a comment in the code.